nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
+            cat_match = re.match(r'^(\d+)_(\d+)$', item)
+            if cat_match:
+                item = int(cat_match.group(1)) * 100 + int(cat_match.group(2))
+
+            num = int(item)
+            if num >= 100000:
+                categories.append(num - 100000)
+            elif num >= 1000:
+                categories.extend([x - 100000 for x in TORZNAB_CATEGORY_MAP.get(num, [])])
+            elif num > 0:
+                categories.append(num)
+
+        # Check if any of the specified categories 'stuck'.
+        if not categories:
+            raise EmptySearchError('Categories {} guaranteed to produce empty result'.format(cat))
+    return categories
+
+def get_tag_filter(tag):
+    include_tags = []
+    exclude_tags = []
+    if tag:
+        for item in tag.split(','):
+            if item[0] == '!':
+                exclude_tags.append(item[1:].lower())
+            else:
+                include_tags.append(item.lower())
+    for incl in include_tags:
+        if incl in exclude_tags:
+            raise EmptySearchError('Tag {} both required and filtered')
+    return (include_tags, exclude_tags)
+
+def torznab_search(api_request):
+    page_size = TORZNAB_RESULTS_PER_PAGE
 
     # Default newznab query params, return empty result if unsupported to prevent clients from accessing the api as a traditional newznab indexer.
-    q = flask.request.args.get('q')
-    season = flask.request.args.get('season')
-    ep = flask.request.args.get('ep')
-    rid = flask.request.args.get('rid')
+    query_term = api_request.args.get('q')
+    query_season = api_request.args.get('season')
+    query_episode = api_request.args.get('ep')
+    query_rageid = api_request.args.get('rid')
```

Comment:
Since you are repeatedly getting items from `api_request.args` assign it to a variable a single time to avoid repeated look-ups:
```python
args = api_request.args
season = args.get('season')
episode = args.get('ep')
...
```